File: examples/ruby/DbIpCity.rb
# Runnable example: IP to City Snapshot (GET /v3.3/download/snapshot/ip/city)
# Parameters for dbIpCity (GET /v3.3/download/snapshot/ip/city):
#   - date (string, required)
require 'date'
require 'whoisfreaks'

WhoisFreaks.configure do |config|
  config.api_key["apiKey"] = "YOUR_API_KEY"   # set once
end

api = WhoisFreaks::DatabasesIPGeolocationApi.new
result = api.db_ip_city((Date.today - 1).to_s)
puts result

File: examples/php/DbIpCity.php
<?php
// Runnable example: IP to City Snapshot (GET /v3.3/download/snapshot/ip/city)
// Parameters for dbIpCity (GET /v3.3/download/snapshot/ip/city):
//   - date (string, required)
require 'vendor/autoload.php';

$config = WhoisFreaks\Configuration::getDefaultConfiguration()
    ->setApiKey("apiKey", "YOUR_API_KEY");  // set once
$api = new WhoisFreaks\Api\DatabasesIPGeolocationApi(new GuzzleHttp\Client(), $config);
$result = $api->dbIpCity((new DateTime("yesterday"))->format("Y-m-d"));
print_r($result);
